[AY1920S1-CS2113T-W13-1] Coach Manager

Sfloydzy opened the pull request:
@danisheddie @eujingsen @NotTheRealEdmund

flxffy commented on src/main/java/duke/module/CategoryList.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    public CategoryList(final String subMenuDescription, final String menu) {
+        super(subMenuDescription, menu);
+    }
```
Why is your constructor at the end of your class?

flxffy commented on src/main/java/duke/module/MyMenu.java:
```diff
@@ -0,0 +1,59 @@
+//    public void trainingProgramHeading() {
+//        System.out.flush();
+//        System.out.println("TRAINING PROGRAM: \n");
+//    }
```
If you don't need it, don't leave it.

flxffy commented on src/main/java/duke/module/Goal.java:
```diff
@@ -0,0 +1,125 @@
+            return "There is no goal of the day";
+        } else {
+            return message;
+        }
```
Flow could be better.

flxffy replied:
This conditional path could flow better for readability. Consider setting the happy path first.

flxffy commented on src/main/java/duke/module/Lesson.java:
```diff
@@ -0,0 +1,132 @@
+                    }
+                }
+            }
+        }
```
Convoluted and unnecessary nesting. Look through the Map documentation and you'll find a better way to do this.

flxffy commented on src/main/java/duke/module/Reminder.java:
```diff
@@ -0,0 +1,92 @@
+        System.out.println("Today's date " + todayDate);
+        System.out.println("Saved date: " + endDate);
+        return todayDate;
+    }
```
Violating single responsibility principle.

flxffy replied:
Violating single responsibility principle. Think about why this is so

flxffy commented on src/main/java/duke/sports/ManageStudents.java:
```diff
@@ -0,0 +1,168 @@
+        } catch (IndexOutOfBoundsException e) {
+            if (getStudentListSize() == 0) {
+                System.err.println("Oops! "
+                        + "You only have no student in the list!");
```
Weird phrasing.

flxffy commented on src/main/java/duke/sports/MyPlan.java:
```diff
@@ -0,0 +1,374 @@
+     /**
+      * Represents the list for the current number of plans saved.
+      */
+    private ArrayList<String> toc = new ArrayList<>();
```
What is toc?

flxffy replied:
What is toc? Use a more descriptive variable name.

flxffy commented on src/main/java/duke/sports/MyPlan.java:
```diff
@@ -0,0 +1,374 @@
+      * @throws FileNotFoundException if file is not found.
+      */
+    public MyPlan() throws FileNotFoundException {
+        filePath = ".\\src\\main\\java\\duke\\data\\plan.txt";
```
Windows file path? Look at your project requirements, it should be cross-platform compatible.

flxffy replied:
Windows file path? Are you sure this works on all platforms? Refer to the [project constraints](https://nuscs2113-ay1920s1.github.io/website/admin/project-constraints.html), your product should be cross-platform compatible.

flxffy commented on src/main/java/duke/sports/MyPlan.java:
```diff
@@ -0,0 +1,374 @@
+                                + "finalize the plan or look at current list.");
+                    }
+                }
+            }
```
Why are you scanning for input here?

flxffy commented on src/main/java/duke/task/Item.java:
```diff
@@ -0,0 +1,155 @@
+    /**
+     * status: whether Item is complete or not.
+     */
+    private Boolean status;
```
Your variable name is not reflective of what it is meant for.

flxffy replied:
Your variable name isn't descriptive nor indicative of what it is meant for. Refer to the coding style for a more appropriate name.

flxffy commented on src/main/java/duke/task/Item.java:
```diff
@@ -0,0 +1,155 @@
+     *
+     * @return String date of the Item
+     */
+    public String getDate() {
+        return "";
+    }
```
Empty string??

flxffy replied:
Why is the getter returning an empty string?

flxffy commented:
A lot more refactoring needs to be done.

Things to note:
1. There are a few variables that should be constants but are not.
2. `Controller` is well done.
3. Look up default constructors. Don't leave empty constructors that don't do anything.
4. Good use of interfaces in `Details`.
5. Code that have commented out - do something about that.
6. Conditional paths - in some cases, the flow could be better. Choose the happy path when it makes sense.
7. Read the docs!! Data structures offered in Java are pretty robust and already offer a lot of what you are trying to implement. 
8. Why do you have tests from Duke, and dummy tests?

flxffy reviewed (commented):
ignore this review

flxffy commented on src/main/java/duke/module/CategoryList.java:
```diff
@@ -0,0 +1,46 @@
+    public CategoryList(final String subMenuDescription, final String menu) {
+        super(subMenuDescription, menu);
+    }
+
```
Why is the constructor at the end of the class?

flxffy commented on src/main/java/menu/MyMenu.java:
```diff
@@ -0,0 +1,41 @@
+//    public void trainingProgramHeading() {
+//        System.out.flush();
+//        System.out.println("TRAINING PROGRAM: \n");
+//    }
```
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

flxffy commented on src/main/java/duke/module/Lesson.java:
```diff
@@ -0,0 +1,132 @@
+                        message += str + "\n";
+                    }
+                }
+            }
```
This logic is convoluted and uses unnecessary nesting. Look through the Map documentation and you'll find a better way to do this!

flxffy commented on src/main/java/duke/sports/MyPlan.java:
```diff
@@ -0,0 +1,374 @@
+            while (true) {
+                Scanner sc = new Scanner(System.in);
+                if (sc.hasNextLine()) {
+                    String input = sc.nextLine();
```
Should `MyPlan` be scanning user input? This is violating one of the design principles we went through last week.

flxffy reviewed (commented):
A lot more refactoring needs to be done.

Things to note:
1. There are a few variables that should be constants but are not.
2. `Controller` is well done.
3. Look up default constructors. Don't leave empty constructors that don't do anything.
4. Good use of interfaces in `Details`.
5. Code that have been commented out - those can be deleted.
6. Conditional paths - in some cases, the flow could be better. Choose the happy path when it makes sense.
7. Read the docs!! Data structures offered in Java are pretty robust and already offer a lot of what you are trying to implement. 
8. Why do you have tests from Duke, and dummy tests?

Commit:
Make Ui handle all the printing for MyPlan according to what aihui wants

Commit:
Checkstyle maintenance for Ui

eujingsen committed:
Merge pull request #194 from NotTheRealEdmund/master

Creating ParserGoal

Commit:
Add in Schedule inside ParserCommand

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Create ParserSchedule

Commit:
Checkstyle for ParserSchedule

Commit:
Add weekly and monthly schedule

eujingsen committed:
Merge pull request #195 from NotTheRealEdmund/master

Add in Schedule inside ParserCommand

Commit:
Bug fix: Add break for switch statements

Commit:
Add a few command classes

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Remove unnecessary code

eujingsen committed:
Merge pull request #196 from NotTheRealEdmund/master

Add a few command classes

eujingsen committed:
Added MyPlan Functionalities

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

eujingsen committed:
MyPlan: Minor changes

Commit:
Merge pull request #197 from eujingsen/master

MyPlan: Added functionalities

Sfloydzy committed:
Merge pull request #26 from AY1920S1-CS2113T-W13-1/master

Merge Updates

eujingsen committed:
Fix error

Commit:
Merge pull request #198 from eujingsen/master

Fix error

Commit:
Checkstyle for ParserCommand

Commit:
Update ParserCommand.java

Sfloydzy committed:
Merge pull request #199 from NotTheRealEdmund/master

Checkstyle for ParserCommand

Sfloydzy committed:
Merge pull request #27 from AY1920S1-CS2113T-W13-1/master

Merge update

danisheddie committed:
Handled merge conflicts

danisheddie committed:
Resolving checkstyle errors

danisheddie committed:
Checkstyle spelling errors

danisheddie committed:
Handling Checkstyle errors

danisheddie committed:
Handling Checkstyle errors

Sfloydzy committed:
Merge pull request #200 from danisheddie/ManageStudents

[CS2113T-W13-1]-danisheddie- Checkstyle edits

Commit:
Checkstyle for Command done

Commit:
Update package-info.java

danisheddie committed:
Merge pull request #201 from NotTheRealEdmund/master

Checkstyle for Command done

danisheddie committed:
handling checkstyle errors

danisheddie committed:
Removed redundant parameter.

Sfloydzy committed:
Merge pull request #202 from danisheddie/ManageStudents

[CS2113T-W13-1]-danisheddie-Checkstyle Errors

Commit:
Un-commented launcher

Commit:
Merge https://github.com/AY1920S1-CS2113T-W13-1/main

# Conflicts:
#	src/main/java/duke/Duke.java

Sfloydzy committed:
Merge pull request #28 from AY1920S1-CS2113T-W13-1/master

Merge update

Sfloydzy committed:
Merge pull request #203 from NotTheRealEdmund/master

Un-commented launcher

Sfloydzy committed:
Merge pull request #29 from AY1920S1-CS2113T-W13-1/master

Merge update

danisheddie committed:
Created a travis yml file

danisheddie committed:
Bug fix to run code.

eujingsen committed:
Correct error

Commit:
Merge pull request #205 from eujingsen/master

Correct error in MyPlan

Commit:
Merge pull request #204 from danisheddie/Travis

[CS2113T-W13-1]-danisheddie-Installed Travis

eujingsen committed:
Fix MyPlan functionalities

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

danisheddie committed:
editted travis file

Sfloydzy committed:
Merge pull request #206 from danisheddie/Travis

Editted travis file

danisheddie committed:
Renaming Ui to CLIView and changing capitalisation of Parser to parser

danisheddie committed:
Settle merge conflicts

danisheddie committed:
Setting up travis

danisheddie committed:
Uncommented javaFX

danisheddie committed:
Deleted unecessary test

danisheddie committed:
Fixing test errors

danisheddie committed:
Merge pull request #207 from danisheddie/ManageStudents

[CS2113T-W13-1]-danisheddie-Renaming Ui and Parser

danisheddie committed:
Added Travis Badge

danisheddie committed:
Checkstyle Handling

Sfloydzy committed:
Merge pull request #30 from AY1920S1-CS2113T-W13-1/master

Merge Update

danisheddie committed:
Handling Checkstyle errors

danisheddie committed:
Handling Checkstyle errors

Sfloydzy committed:
Fixed all checkstyles in code

Sfloydzy committed:
Merge pull request #208 from Sfloydzy/master

Fixed all checkstyle errors in all files

danisheddie committed:
Handling merge conflicts

Sfloydzy committed:
Touched up some code

danisheddie committed:
Enable Travis to check checkstyle

Sfloydzy committed:
Merge pull request #210 from danisheddie/ManageStudents

[CS2113T-W13-1]-danisheddie-Travis Set-up

danisheddie committed:
Refactoring the classes into their respective packages

danisheddie committed:
Merge pull request #211 from danisheddie/ManageStudents

[CS2113T-W13-1]-danisheddie-Refactoring packages

danisheddie committed:
Created a package for manageStudents consisting of ManageStudent class and MyStudent class

Sfloydzy committed:
Merge https://github.com/AY1920S1-CS2113T-W13-1/main into tempBranch

Sfloydzy committed:
Merge branch 'tempBranch' to master to fix conflicts

Sfloydzy committed:
Reformatted parser for schedule.

Added back option to schedule and looping menu so that you remain in that
section until you decide to leave.

Fix error handling for nullpointer exception in daily schedule. "List"
inside schedule remains completely empty all the time and is not updated.

Sfloydzy committed:
Completely commented out loading of text file from schedule.
Does not work and is interrupting the jar file from operating.

Re-examine why so many objects need to be loaded from the beginning as well?

danisheddie committed:
Merge pull request #213 from Sfloydzy/master

Fixes and Improvements for Schedule

danisheddie committed:
Merge remote-tracking branch 'upstream/master' into ManageStudents

danisheddie committed:
Minor edits for ManageStudent commands

danisheddie committed:
renaming managestudentspackage to students

eujingsen committed:
Added methods for parserTrainingPlan: in progress

Sfloydzy committed:
Merge pull request #214 from danisheddie/ManageStudents

[CS2113T-W13-1]-danisheddie-Edits for ManageStudent command and CLIView

Commit:
Refactoring: Move all Goal and Lesson methods to their respective Command classes.

Commit:
Trying to pass Travis

Commit:
Trying to pass Travis

Commit:
Trying to pass Travis

Commit:
Trying to pass Travis

Commit:
Make code RepoSense-compatible

Commit:
Trying to pass Travis

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

eujingsen committed:
Merge pull request #215 from NotTheRealEdmund/master

Refactoring: Move all Goal and Lesson methods to their respective Command classes.

eujingsen committed:
Fix merge conflict + Travis checkstyle

Sfloydzy committed:
Merge pull request #32 from AY1920S1-CS2113T-W13-1/master

Merge updates

Sfloydzy committed:
Updated project portfolio

Sfloydzy committed:
Testing line

Sfloydzy committed:
Merge pull request #217 from Sfloydzy/master

Testing CI

Sfloydzy committed:
Merge branch 'master' into master

danisheddie committed:
Code refactoring and commmand edits for Manage Students

danisheddie committed:
handling checkstye errors

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

Sfloydzy committed:
Merge pull request #218 from danisheddie/ManageStudents

Code refactoring and commmand edits for Manage Students

eujingsen committed:
Update ParserSchedule.java

eujingsen committed:
Update ParserSchedule.java

Sfloydzy committed:
Merge pull request #219 from eujingsen/master

Fix travis checkstyle + adjustments to code

Sfloydzy committed:
Merge pull request #33 from AY1920S1-CS2113T-W13-1/master

Merge update

Sfloydzy committed:
Update Project portfolio

build.gradle updated as well

Sfloydzy committed:
Merge pull request #220 from Sfloydzy/master

Update project portfolio

Sfloydzy committed:
Update SfloydzyPortfolio.adoc

Removed small errors with contributions

danisheddie committed:
Updated portfolio page

danisheddie committed:
Merge pull request #221 from danisheddie/ManageStudents

Updated portfolio page

Sfloydzy committed:
Update SfloydzyPortfolio.adoc

danisheddie committed:
Minor edit to the format

danisheddie committed:
Edited format

Sfloydzy committed:
Update SfloydzyPortfolio.adoc

eujingsen committed:
Update Portfolio

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

Sfloydzy committed:
Update SfloydzyPortfolio.adoc

eujingsen committed:
Update eujingsenPortfolio.adoc

eujingsen committed:
Update eujingsenPortfolio.adoc

Commit:
Fix nullpointerexception error

eujingsen committed:
Merge pull request #222 from NotTheRealEdmund/master

Fix nullpointerexception error

Commit:
Update edmundPortfolio.adoc

Commit:
Add files via upload

Commit:
Update edmundPortfolio.adoc

Commit:
Trying to pass Travis

Commit:
Merge branch 'master' of https://github.com/NotTheRealEdmund/main

danisheddie committed:
Merge pull request #223 from NotTheRealEdmund/master

Update edmundPortfolio.adoc

Sfloydzy committed:
Update SfloydzyPortfolio.adoc

danisheddie committed:
Editteed portfolio

danisheddie committed:
Merge branch 'master' into ManageStudents

danisheddie committed:
Merge pull request #224 from danisheddie/ManageStudents

Edited portfolio

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

eujingsen committed:
Nullexception Error fixed

danisheddie committed:
Editted add and find command for manage students

danisheddie committed:
Merge pull request #225 from danisheddie/ManageStudents

Editted add and find command for manage students

eujingsen committed:
Added MyPlan functionalities + NullPointerException fixed

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

eujingsen committed:
Update input.txt

danisheddie committed:
[REfactoring] created interface for student and student list.

danisheddie committed:
Changed class name from Mystudent to student

danisheddie committed:
Refactored student classes

danisheddie committed:
Merge pull request #228 from danisheddie/ManageStudents

Added interface to student classes

danisheddie committed:
Merge pull request #227 from eujingsen/master

MyPlan : added functionalities

Commit:
Update edmundPortfolio.adoc

eujingsen committed:
Merge pull request #229 from NotTheRealEdmund/master

Update edmundPortfolio.adoc

Sfloydzy committed:
Merge pull request #34 from AY1920S1-CS2113T-W13-1/master

Merge Updates

Commit:
Update edmundPortfolio.adoc

Commit:
Fix bug from PE dry run

Commit:
Merge https://github.com/NotTheRealEdmund/main

Commit:
Fix bug found in PE dry run, create new txt file if it doesn't exist/is deleted

Commit:
Trying to pass Travis

Commit:
Fix bug from PE dry run, add exception to handle InputMismatchException

Commit:
Trying to pass Travis

Commit:
Trying to pass Travis

Commit:
Trying to pass Travis

eujingsen committed:
Merge pull request #230 from NotTheRealEdmund/master

Update edmundPortfolio.adoc

danisheddie committed:
Changed variable name

danisheddie committed:
Edits to manage students parser

danisheddie committed:
Creatd a storage for student list

Sfloydzy committed:
Merge pull request #264 from danisheddie/ManageStudents

Created a storage for student list

Sfloydzy committed:
Merge pull request #35 from AY1920S1-CS2113T-W13-1/master

Merge Update

Sfloydzy committed:
Implemented json file loading and saving templates

Further elaboration needed, unsafe operation according to gradle

danisheddie committed:
Merge pull request #265 from Sfloydzy/master

Added json file loading

danisheddie committed:
Changed ManageStudent name to StudentList

danisheddie committed:
Created Student Progress List class and Progress List interface

danisheddie committed:
Editted bugs in stroage

danisheddie committed:
Added command to view particular student details

danisheddie committed:
Changed parameter and type for progress

danisheddie committed:
Edits to manageStudent Parser

danisheddie committed:
Added progress parser

danisheddie committed:
student list text file

danisheddie committed:
Checkstyle edits

Sfloydzy committed:
Added option to select months in calender

Menu system included inside the calender for months

Sfloydzy committed:
Removed CheckStyle Errors

Sfloydzy committed:
Merge pull request #266 from danisheddie/ManageStudents

Added progress class

Sfloydzy committed:
Merge pull request #268 from Sfloydzy/master

Added option to select months in calender

Sfloydzy committed:
Merge pull request #36 from AY1920S1-CS2113T-W13-1/master

Merge Updates

Sfloydzy committed:
Updated schedule of the month functions

Commit:
Add lines

eujingsen committed:
Merge pull request #269 from NotTheRealEdmund/master

Add lines

danisheddie committed:
Added progress interface.

danisheddie committed:
Editted navigability for managestuents

danisheddie committed:
Editted heading for managestudents

danisheddie committed:
Change launcher name to run program

danisheddie committed:
Edits to student list feature

danisheddie committed:
Editted type

Sfloydzy committed:
Load table format for the day view in the calender

Saved data now has to be loaded from the resource file
get the info with getResourceasStream() then parse the buffer
stream as a string

Sfloydzy committed:
Merge branch 'master' into master

Sfloydzy committed:
Merge pull request #271 from Sfloydzy/master

Added Table View for schedule, fixed jar loading, implemented json files

Sfloydzy committed:
Merge branch 'master' into ManageStudents

Sfloydzy committed:
Merge pull request #270 from danisheddie/ManageStudents

Manage students - Edited navigability

Sfloydzy committed:
Merge pull request #37 from AY1920S1-CS2113T-W13-1/master

Merge Updates

Sfloydzy committed:
Revert "Manage students - Edited navigability"

Sfloydzy committed:
Merge pull request #272 from AY1920S1-CS2113T-W13-1/revert-270-ManageStudents

Revert "Manage students - Edited navigability"

Sfloydzy committed:
Merge pull request #38 from AY1920S1-CS2113T-W13-1/master

revert changes

Sfloydzy committed:
Fixed some bugs

eujingsen committed:
MyPlan: Functionalities added(Still in progress)

Sfloydzy committed:
List view for all created schedule items is now possible

eujingsen committed:
MyPlan: Save function WIP

danisheddie committed:
Added student progres functionality and editted navigability for Managing Students

Sfloydzy committed:
Merge pull request #274 from danisheddie/ManageStudents

Student Progress & Navigability for Manage Students

eujingsen committed:
Fix travis(Small error)

Sfloydzy committed:
Logger works together with JAR file saving

Sfloydzy committed:
Made changes to ScheduleStorage to validate files

Sfloydzy committed:
Merge branch 'master' into master

Sfloydzy committed:
Fixed CheckStyle Errors

Editted Portfolio

Sfloydzy committed:
Fixed Saving and Added Logger #275 from Sfloydzy/master

Fixed saving and added ApacheLogger properly

danisheddie committed:
Added progress command in parser

danisheddie committed:
Added progress option in cli view

danisheddie committed:
Removed unused class

Sfloydzy committed:
Add class to the table in Schedule

Able to add class to the schedule, in addition dates which do not have classes scheduled will be updated in the "list command".

Sfloydzy committed:
Tidy Code

danisheddie committed:
Createt test for student

danisheddie committed:
Student progress edit

danisheddie committed:
Added test methods in student test

danisheddie committed:
Merge pull request #276 from Sfloydzy/master

Ability to add classes to a specific day

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

danisheddie committed:
Removed unnecesary code

Sfloydzy committed:
Date validation for table

Sfloydzy committed:
Added Tests and changed PPP

Sfloydzy committed:
Merge pull request #277 from danisheddie/ManageStudents

Added testings and edited progress list for students

Sfloydzy committed:
Merge pull request #273 from eujingsen/master

MyPlan: Functionalities added(Still in progress)

Sfloydzy committed:
Pushing files which did not push before

Sfloydzy committed:
Merge pull request #278 from Sfloydzy/master

Merge ppp updates and tests

danisheddie committed:
Edited save file

danisheddie committed:
removed redundant code

danisheddie committed:
Checkstyle edit

danisheddie committed:
Added author and removed redundant code

danisheddie committed:
Editted ManageStudent parser

danisheddie committed:
Checkstyle edit

danisheddie committed:
Removed unused options

danisheddie committed:
Added authorship

eujingsen committed:
MyPlan fixed save function

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

eujingsen committed:
Merge pull request #279 from danisheddie/ManageStudents

ManageStudents code edits

Commit:
Merge pull request #280 from eujingsen/master

MyPlan fixed functionalities

danisheddie committed:
Fixed bug for adding student progress

danisheddie committed:
Fixed bug for adding progress

Sfloydzy committed:
Final Changes to gradle file

danisheddie committed:
Edit checkstyle

Sfloydzy committed:
Merge pull request #282 from Sfloydzy/master

Final Changes to gradle file

Sfloydzy committed:
Merge pull request #281 from danisheddie/ManageStudents

Final edit/bug fixing

Commit:
Final changes

Commit:
Merge pull request #284 from NotTheRealEdmund/master

Final changes

eujingsen committed:
File directory fixed

eujingsen committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge pull request #285 from eujingsen/master

Txt file directory for JAR fixed

danisheddie committed:
Final bug edit

danisheddie committed:
Change to single class imports

danisheddie committed:
Merge pull request #286 from danisheddie/ManageStudents

Final bug edit
